### PR TITLE
Add documentation for the transactable module private preview

### PR DESCRIPTION
**Pull request by jlian**
This PR adds the instructions that partners need to follow for the transactable edge module private preview. No source code updates.

## Azure IoT Edge PR checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines and Best Practices
- [x] I have read the [contribution guidelines](https://github.com/azure/iotedge#contributing).
- [x] Title of the pull request is clear and informative.
- [x] Description of the pull request includes a concise summary of the enhancement or bug fix.

### Testing Guidelines

No code changes in this PR

- [ ] Pull request includes test coverage for the included changes.
- Description of the pull request includes 
	- [ ] concise summary of tests added/modified
	- [ ] local testing done.  

### Draft PRs
- Open the PR in `Draft` mode if it is:
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.

_Note: We use the kodiakhq bot to merge PRs once the necessary checks and approvals are in place. When it merges a PR, kodiakhq converts the PR title to the commit title, PR description to the commit description, and squashes all the commits in the PR to a single commit. The net effect is that entire PR becomes a single commit. Please follow the best practices mentioned [here](https://chris.beams.io/posts/git-commit/#:~:text=The%20seven%20rules%20of%20a%20great%20Git%20commit,what%20and%20why%20vs.%20how%20For%20example%3A%20) for the PR title and description_

### CODE REVIEW

File: edge-modules/TransactableModuleSample/README.md [hunk 1]
```diff
@@ -4,9 +4,9 @@
...
 
-Here's a list of features that are supported. We have tested each of the features, but we haven't tested all the possible combinations in every possible order. Please try them out! If you find any bugs beyond, please send us an email at emailedgetransactablemodulespreview@service.microsoft.com.
+Here's a list of features that are supported. We have tested each of the support features, but we haven't tested all the possible combinations in every possible order. Please try them out! If you find any bugs beyond, please send us an email at emailedgetransactablemodulespreview@service.microsoft.com. Please don't attempt any of the unsupported scenarios like moving your IoT hub resource.
```
Comment (varunpuranik) [inline edge-modules/TransactableModuleSample/README.md:9]:
```suggestion
Here's a list of features that are supported. We have tested each of the support features, but we haven't tested all the possible combinations in every possible order. Please try them out! If you find any bugs beyond, please send us an email at emailedgetransactablemodulespreview@service.microsoft.com. Please don't attempt any of the unsupported scenarios like moving your IoT Hub resource. 
```

**Review by varunpuranik — APPROVED** [review]

**Review by AarathiN — APPROVED** [review]

File: edge-modules/TransactableModuleSample/README.md [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+
+This section shows steps to create a new transactable Edge module offer. For the private preview, the offer *must be live* to be deployable to an Edge device. To hide the offer from the public, ensure that each *plan* you create is hidden.
+
+1. If you haven't done so, email edgetransactablemodulespreview@service.microsoft.com your seller ID. Wait for a positive confirmation before proceeding.
+1. Visit [Partner Center - Commercial Marketplace Overview](https://partner.microsoft.com/en-us/dashboard/commercial-marketplace/overview)
+1. Click **+ New offer** and select **IoT Edge module**
+1. Give the new offer a descriptive name and alias, then click **Create**
+1. In the refreshed page, click **Plan overview** and **+ Create new plan**
+1. Give the plan a descriptive name and ID, then click **Create**
+1. In **Pricing and availability**, adjust the **available markets**, set your **pricing** (we recommend using the smallest possible amount, like $0.01/module/hour for private preview), ande **hide** the plan
```
Comment (jlian) [inline edge-modules/TransactableModuleSample/README.md]:
```suggestion
1. In **Pricing and availability**, adjust the **available markets**, set your **pricing**. **To minimize any financial impact, please use the smallest possible amount, like $0.01/module/hour. Higher pricing without a justification will lead to delay or rejection during certification**. 
1. **Hide** the plan
```

Comment (jlian) [reply edge-modules/TransactableModuleSample/README.md]: @AarathiN

File: edge-modules/TransactableModuleSample/README.md [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+
+This section shows steps to create a new transactable Edge module offer. For the private preview, the offer *must be live* to be deployable to an Edge device. To hide the offer from the public, ensure that each *plan* you create is hidden.
+
+1. If you haven't done so, email edgetransactablemodulespreview@service.microsoft.com your seller ID. Wait for a positive confirmation before proceeding.
+1. Visit [Partner Center - Commercial Marketplace Overview](https://partner.microsoft.com/en-us/dashboard/commercial-marketplace/overview)
+1. Click **+ New offer** and select **IoT Edge module**
+1. Give the new offer a descriptive name and alias, then click **Create**
+1. In the refreshed page, click **Plan overview** and **+ Create new plan**
+1. Give the plan a descriptive name and ID, then click **Create**
+1. In **Pricing and availability**, adjust the **available markets**, set your **pricing** (we recommend using the smallest possible amount, like $0.01/module/hour for private preview), and **hide** the plan
```
Comment (jlian) [inline edge-modules/TransactableModuleSample/README.md:95]:
```suggestion
1. In **Pricing and availability**, adjust the **available markets**, set your **pricing**. **To minimize any financial impact, please use the smallest possible amount, like $0.01/module/hour. Higher pricing without a justification will lead to delay or rejection during certification**. 
1. **Hide** the plan
```

**Review by varunpuranik — APPROVED** [review]

**Pull request merged**